src/types.d.ts

```diff
@@ -0,0 +1,414 @@
+/**
```

Comment:
Interfaces/types move to this file. A lot of new types were added, some types were improved.